src/rgw/driver/rados/rgw_data_sync.h

```diff
@@ -340,9 +338,9 @@ struct RGWDataSyncEnv {
 
 // pretty ostream output for `radosgw-admin bucket sync run`
 template<typename ...T>
-void pretty_print(const RGWDataSyncEnv* env, T&& ...t) {
+void pretty_print(const RGWDataSyncEnv* env, fmt::format_string<T...> fmt, T&& ...t) {
   if (unlikely(!!env->ostr)) {
-    fmt::print(*env->ostr, std::forward<T>(t)...);
+    fmt::print(*env->ostr, fmt, std::forward<T>(t)...);
```

Comment:
this doesn't seem to work with fmt-devel 8.1.1 (fedora 36):
```
In file included from ceph/src/rgw/driver/rados/rgw_data_sync.h:8,
                 from ceph/src/rgw/driver/rados/rgw_data_sync.cc:13:
/usr/include/fmt/ostream.h: In instantiation of ‘void fmt::v8::print(std::basic_ostream<Char>&, const S&, Args&& ...) [with S = basic_format_string<char, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&>; Args = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}; Char = char]’:
ceph/src/rgw/driver/rados/rgw_data_sync.h:343:15:   required from ‘void pretty_print(const RGWDataSyncEnv*, fmt::v8::format_string<T ...>, T&& ...) [with T = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}; fmt::v8::format_string<T ...> = fmt::v8::basic_format_string<char, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&>]’
ceph/src/rgw/driver/rados/rgw_data_sync.cc:5779:17:   required from here
/usr/include/fmt/ostream.h:130:28: error: no matching function for call to ‘to_string_view(const fmt::v8::basic_format_string<char, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&>&)’
  130 |   vprint(os, to_string_view(format_str),
      |              ~~~~~~~~~~~~~~^~~~~~~~~~~~
```
full log https://paste.centos.org/view/raw/3229396b

Reply:
the error seems to be related to the rgw_obj_key ostream operator

Reply:
@cbodley #49479 should fix this. sorry for the inconvenience.